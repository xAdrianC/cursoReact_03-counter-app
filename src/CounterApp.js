import {useState} from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = (e) => {
        setCounter(counter + 1)
    }

    const handleSubtract = (e) => {
        setCounter(counter - 1)
    }

    const handleReset = (e) => {
        setCounter(value)
    }


    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSubtract}>-1</button>
        </>
        
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    
}

export default CounterApp;