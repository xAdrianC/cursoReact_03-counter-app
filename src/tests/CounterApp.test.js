import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import CounterApp from "../CounterApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';


describe('Pruebas en primeraApp', () => {

    let wrapper = shallow(<CounterApp value={5}/>);

    beforeEach(() => {
        const value = 5;
        wrapper = shallow(<CounterApp value={value}/>);
    })

    test('debe mostrar <CounterApp /> correctamente ', () => {
        const value = 5;
        wrapper = shallow(<CounterApp value={value}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar como valor inicial 100 ', () => {
        const value = 100;
        wrapper = shallow(<CounterApp value={value}/>);

        const valueCounter = wrapper.find('h2').text();
        expect(Number(valueCounter)).toBe(100);
    });

    test('debe incrementar con el +1 ', () => {
        

        const btn1 = wrapper.find('button').at(0);
        btn1.simulate('click');

        const valueCounter = wrapper.find('h2').text();
        expect(Number(valueCounter)).toBe(6);
    });

    test('debe decrementar con el -1 ', () => {

        const btn3 = wrapper.find('button').at(2);
        btn3.simulate('click');

        const valueCounter = wrapper.find('h2').text();
        expect(Number(valueCounter)).toBe(4);
    });

    test('debe colocar el valor por defecto con rst', () => {
        const value = 105;
        wrapper = shallow(<CounterApp value={value}/>);

        let btn1 = wrapper.find('button').at(0);
        btn1.simulate('click');
        btn1 = wrapper.find('button').at(0);
        btn1.simulate('click');
        let btnRst = wrapper.find('button').at(1);
        btnRst.simulate('click');


        const valueCounter = wrapper.find('h2').text();
        expect(Number(valueCounter)).toBe(value);
    })
    
    
    
});