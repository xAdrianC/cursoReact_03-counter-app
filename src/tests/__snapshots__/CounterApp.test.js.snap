// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primeraApp debe mostrar <CounterApp /> correctamente  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    5
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
